jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/config/JmxScraperConfig.java

```diff
@@ -0,0 +1,244 @@
+   * OTel SDK or SSL/TLS stack initialization. This allows to override JVM system properties using
+   * user-providded configuration and also to set standard OTel SDK configuration.
+   */
+  public void propagateSystemProperties() {
+    for (Map.Entry<Object, Object> entry : properties.entrySet()) {
+
+      String key = entry.getKey().toString();
+      String value = entry.getValue().toString();
+      if (key.startsWith("otel.")
+          || key.startsWith("javax.net.ssl.keyStore")
+          || key.startsWith("javax.net.ssl.trustStore")) {
+        System.setProperty(key, value);
+      }
+    }
+  }
+
+  private static int getProperty(Properties properties, String key, int defaultValue)
+      throws ConfigurationException {
+    String propVal = properties.getProperty(key);
+    if (propVal == null) {
+      return defaultValue;
+    }
+    try {
+      return Integer.parseInt(propVal);
+    } catch (NumberFormatException e) {
+      throw new ConfigurationException("Failed to parse " + key, e);
+    }
+  }
+
+  /**
+   * Similar to getProperty(key, defaultValue) but sets the property to default if not in object.
+   */
+  private static String getAndSetPropertyIfUndefined(
+      Properties properties, String key, String defaultValue) {
+    String propVal = properties.getProperty(key, defaultValue);
+    if (propVal.equals(defaultValue)) {
+      properties.setProperty(key, defaultValue);
+    }
+    return propVal;
+  }
+
+  private static int getAndSetPropertyIfUndefined(
+      Properties properties, String key, int defaultValue) throws ConfigurationException {
+    int propVal = getProperty(properties, key, defaultValue);
+    if (propVal == defaultValue) {
+      properties.setProperty(key, String.valueOf(defaultValue));
+    }
+    return propVal;
+  }
+
+  /** Will determine if parsed config is complete, setting any applicable values and defaults. */
+  private static void validateConfig(JmxScraperConfig config) throws ConfigurationException {
+    if (isBlank(config.serviceUrl)) {
+      throw new ConfigurationException(SERVICE_URL + " must be specified.");
+    }
+
+    if (isBlank(config.customJmxScrapingConfigPath) && config.targetSystems.isEmpty()) {
+      throw new ConfigurationException(
+          CUSTOM_JMX_SCRAPING_CONFIG + " or " + TARGET_SYSTEM + " must be specified.");
+    }
+
+    if (!config.targetSystems.isEmpty()
+        && !AVAILABLE_TARGET_SYSTEMS.containsAll(config.targetSystems)) {
+      throw new ConfigurationException(
+          String.format(
+              "%s must specify targets from %s", config.targetSystems, AVAILABLE_TARGET_SYSTEMS));
+    }
+
+    if (isBlank(config.otlpExporterEndpoint)
+        && (!isBlank(config.metricsExporterType)
+            && config.metricsExporterType.equalsIgnoreCase("otlp"))) {
+      throw new ConfigurationException(OTLP_ENDPOINT + " must be specified for otlp format.");
+    }
```

Comment:
Can we default the endpoint to the same default that the agent uses? OTLP http+protobuf to localhost:4318? Seems like a reasonable/sane default to me.

Reply:
Should now be fixed by https://github.com/open-telemetry/opentelemetry-java-contrib/pull/1445/commits/2c95782292d06eb29c41f266466e3648ca025fc8